src/sema/file.rs

```diff
@@ -71,6 +79,16 @@ impl File {
             self.line_starts[line_no - 1] + column_no
         }
     }
+
+    pub fn file_name(&self) -> String {
+        #[cfg(not(windows))]
+        let res = format!("{self}").split('/').last().unwrap().to_string();
+
+        #[cfg(windows)]
+        let res = format!("{self}").split('\\').last().unwrap().to_string();
+
+        res
```

Comment:
Don't do your own file name handling, better to the standard library for that:
 ```rust
format!("{}", self.path.filename().unwrap().to_path_buf())
```

Reply:
`path.file_name()` is an `OsStr`, which should be converted to an `str` first as `OsStr` doesn't implement `display` or `to_path_buf`. This what I was doing in `self.path.file_name().unwrap().to_str().unwrap()`. 

I think doing it by hand is better. (`display()` is first implemented for windows and other Os', then it's parsed accordingly)

Reply:
You're right, `.to_path_buf()` is not implemented and it's not great either to convert to another struct.

I still think you can avoid doing the parsing by hand. This time I've tested it in on branch.

```rust
    pub fn file_name(&self) -> String {
        self.path.file_name().unwrap().to_string_lossy().into()
    }
```

Reply:
Thanks!